compiler/pathutils.nim

```diff
@@ -54,21 +62,21 @@ when true:
 
   proc `==`*[T: AnyPath](x, y: T): bool = eqImpl(x.string, y.string)
 
-  proc `/`*(base: AbsoluteDir; f: RelativeFile): AbsoluteFile =
```

Comment:
In my world bugfixes are not a justification to rewrite code in a style you happen to like better.

Reply:
reverted to the previous way (which had a lot of code duplication) and instead using a `postProcessBase` which handles the empty `base` case; the correct behavior would be to throw when base is `empty` as was ruled here https://github.com/nim-lang/Nim/pull/10018#issuecomment-448192956 however this would require more unrelated changes and should be done in separate PR; this PR doesn't change externally visible behavior but at least doesn't "lie" (see https://github.com/nim-lang/Nim/pull/13116#discussion_r365580322)